src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
True, it would mean miners would have to make sure to upgrade before THRESHOLD_LOCKED_IN rather than be given an additional 14 day grace period. However, it would also increase the chance of being compatible with BIP148 nodes by about 14 days, and would potentially increase community support by about that fraction of people. I'm not sure if it's worth the risk, but it is worth considering.